### PR TITLE
Add kme enclave manager

### PR DESCRIPTION
**Pull request by rranjan3**
- Slight modifications in base_listener and tcs_listener to
  include start as a member fucntion
- Add kme enclave manager. Request handler and integration
  with components pending.
- Update kme listener with latest intent(earlier written to
  run as an independent entity)
- Integrate listener into kme enclave

Signed-off-by: Rajeev Ranjan <rajeev2.ranjan@intel.com>

### CODE REVIEW

File: enclave_manager/avalon_enclave_manager/kme/enclave_manager.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+# -------------------------------------------------------------------------
+
+    def _execute_work_order(self, input_json_str, indent=4):
+        """
+        Submits request to KME and retrieves the response
+
+        Parameters :
+            input_json_str - A JSON formatted str of the request to execute.
+        Returns :
+            json_response - A JSON formatted str of the response recieved from
```
Comment (danintel) [inline enclave_manager/avalon_enclave_manager/kme/enclave_manager.py:75]: s/recieved/received/

File: enclave_manager/avalon_enclave_manager/singleton/enclave_manager.py [hunk 2]
```diff
@@ -219,6 +215,8 @@ def _execute_work_order(self, input_json_str, indent=4):
         """
         Submits workorder request to Worker enclave and retrieves the response
 
+        Parameters :
+            input_json_str - A JSON formatted str of the request to execute.
         Returns :
             json_response - A JSON formatted str of the response recieved from
```
Comment (danintel) [inline enclave_manager/avalon_enclave_manager/singleton/enclave_manager.py:221]: s/recieved/received/

File: common/python/listener/base_jrpc_listener.py [hunk 3]
```diff
@@ -163,26 +163,25 @@ def render_POST(self, request):
                 "request {0}: {1}".format(request.path, str(err)))
             return response
 
-# -----------------------------------------------------------------
-# -----------------------------------------------------------------
+    def start(self, host_name, port):
```
Comment (danintel) [inline common/python/listener/base_jrpc_listener.py:166]: I know this is existing code, but could you add a brief python """ docstring  """ comment?

File: enclave_manager/avalon_enclave_manager/kme/enclave_manager.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        WPEs. It terminates only when an exception occurs within start
+        method os the listener instance.
+        """
+        host_name, port = parse_bind_url(
+            self._config["KMEListener"].get("bind"))
+
+        rpc_methods = [
+            self.GetUniqueVerificationKey,
+            self.RegisterWorkorderProcessor,
+            self.PreporocessWorkorder
```
Comment (manojgop) [inline enclave_manager/avalon_enclave_manager/kme/enclave_manager.py:130]: PreporocessWorkorder -> PreProcessWorkOrder

File: enclave_manager/avalon_enclave_manager/kme/enclave_manager.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        ]
+        kme_listener = KMEListener(rpc_methods)
+        kme_listener.start(host_name, port)
+
+
+# -----------------------------------------------------------------
+
+    def GetUniqueVerificationKey(self, **parmas):
```
Comment (manojgop) [inline enclave_manager/avalon_enclave_manager/kme/enclave_manager.py:138]: parmas -> params

File: enclave_manager/avalon_enclave_manager/kme/enclave_manager.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+
+def main(args=None):
+    import config.config as pconfig
+    import utility.logger as plogger
+
+    # parse out the configuration file first
+    tcf_home = os.environ.get("TCF_HOME", "../../../../")
+
+    conf_files = ["singleton_enclave_config.toml"]
```
Comment (manojgop) [inline enclave_manager/avalon_enclave_manager/kme/enclave_manager.py:167]: singleton_enclave_config.toml -> kme_config.toml

File: enclave_manager/avalon_enclave_manager/kme/enclave_manager.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+        self._start_kme_listener()
+
+
+# -------------------------------------------------------------------------
+
+    def _start_kme_listener(self):
+        """
+        This function reads the host_name and port for the listener to
+        bind. It the instantiates the KMEListener and starts it. This
```
Comment (manju956) [inline enclave_manager/avalon_enclave_manager/kme/enclave_manager.py:119]: It the instantiates -> It instantiates

File: enclave_manager/avalon_enclave_manager/kme/enclave_manager.py [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+            self.RegisterWorkorderProcessor,
+            self.PreporocessWorkorder
+        ]
+        kme_listener = KMEListener(rpc_methods)
+        kme_listener.start(host_name, port)
+
+
+# -----------------------------------------------------------------
+
+    def GetUniqueVerificationKey(self, **parms):
```
Comment (manju956) [inline enclave_manager/avalon_enclave_manager/kme/enclave_manager.py:138]: Is this function exposed outside? If its meant to be used internally, you can prepend function with underscore. Same thing applies to other methods as well.

Comment (rranjan3) [reply enclave_manager/avalon_enclave_manager/kme/enclave_manager.py]: Renamed as of now though I guess I might want to move it to the listener. Sounds more logical to have the handlers along side the listener.

**Pull request merged**